src/overlays/actors/ovl_En_Ik/z_en_ik.c

```diff
@@ -132,115 +224,871 @@ static DamageTable D_8092C140 = {
 
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/ovl_En_Ik/func_8092ACFC.s")
+void En_IkUpdateArmorDraw(EnIk* this, GlobalContext* globalCtx) {
+    Gfx* gfxOpa;
+    Gfx* gfxXlu;
+    s32 sp54;
+    EnIkStruct* ptr;
+    s32 i;
 
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/ovl_En_Ik/func_8092ADB4.s")
+    if (this->drawArmorFlags == 3) {
+        sp54 = 0;
 
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/ovl_En_Ik/func_8092AE14.s")
+        OPEN_DISPS(globalCtx->state.gfxCtx);
+        gfxOpa = POLY_OPA_DISP;
+        gfxXlu = POLY_XLU_DISP;
 
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/ovl_En_Ik/func_8092AFB4.s")
+        for (i = 0; i < ARRAY_COUNT(this->unk_550); i++) {
+            ptr = &this->unk_550[i];
+            if (ptr->unk_24 != 0) {
+                Matrix_SetStateRotationAndTranslation(ptr->unk_04.x, ptr->unk_04.y, ptr->unk_04.z, &ptr->unk_1C);
+                Matrix_Scale(0.012f, 0.012f, 0.012f, 1);
 
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/ovl_En_Ik/func_8092AFD4.s")
+                gSPMatrix(gfxOpa++, Matrix_NewMtx(globalCtx->state.gfxCtx),
+                          G_MTX_NOPUSH | G_MTX_LOAD | G_MTX_MODELVIEW);
+                gSPDisplayList(gfxOpa++, ptr->unk_00);
 
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/ovl_En_Ik/func_8092B03C.s")
+                if (D_8092BFA0[i].unk00 != NULL) {
+                    gSPMatrix(gfxXlu++, Matrix_NewMtx(globalCtx->state.gfxCtx),
+                              G_MTX_NOPUSH | G_MTX_LOAD | G_MTX_MODELVIEW);
+                    gSPDisplayList(gfxXlu++, D_8092BFA0[i].unk00);
+                }
+            } else {
+                sp54++;
+            }
+        }
 
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/ovl_En_Ik/func_8092B098.s")
+        if (sp54 == ARRAY_COUNT(this->unk_550)) {
+            this->drawArmorFlags = 5;
+        }
 
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/ovl_En_Ik/func_8092B1B4.s")
+        POLY_XLU_DISP = gfxXlu;
+        POLY_OPA_DISP = gfxOpa;
 
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/ovl_En_Ik/func_8092B46C.s")
+        CLOSE_DISPS(globalCtx->state.gfxCtx);
+    }
+}
 
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/ovl_En_Ik/EnIk_Update.s")
+void EnIk_Draw(Actor* thisx, GlobalContext* globalCtx) {
+    EnIk* this = THIS;
+    Gfx* gfx;
+    Gfx** temp_v1_2;
+    s32 pad;
 
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/ovl_En_Ik/func_8092B900.s")
+    OPEN_DISPS(globalCtx->state.gfxCtx);
 
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/ovl_En_Ik/func_8092B93C.s")
+    func_800B8050(&this->actor, globalCtx, 0);
+    func_800B8118(&this->actor, globalCtx, 0);
+    gfx = POLY_XLU_DISP;
+    gSPDisplayList(&gfx[0], &sSetupDL[6 * 25]);
+    POLY_XLU_DISP = &gfx[1];
+    gfx = POLY_OPA_DISP;
+    gSPDisplayList(&gfx[0], &sSetupDL[6 * 25]);
+    temp_v1_2 = D_8092BFD8[this->actor.params];
+    gSPSegment(&gfx[1], 0x08, temp_v1_2[0]);
+    gSPSegment(&gfx[2], 0x09, temp_v1_2[1]);
+    gSPSegment(&gfx[3], 0x0A, temp_v1_2[2]);
```

Comment:
Optional: this looks like a fake temp, like you can just substitute it out. (although it wouldnt surprise me if I was wrong)

Reply:
I did try, and from what I can see it is a necessary temp (I could set up a decomp.me if you wanted to double check)